pkg/controller/builder/pod_builder.go

```diff
@@ -53,6 +53,9 @@ func BuildActiveGatePodSpecs(instance *v1alpha1.ActiveGate, tenantInfo *dtclient
 		Tolerations:        activeGateSpec.Tolerations,
 		PriorityClassName:  activeGateSpec.PriorityClassName,
 	}
+
+	preparePodSpecImmutableImage(&p, instance)
```

Comment:
And the returned error?

Reply:
Changed the return type and added proper error handling